src/views/StatsView.vue

```diff
@@ -115,19 +118,39 @@ export default {
       return this.seasons.map((season) => {
         return {
           title: season.name,
-          value: season
-        }
+          value: season,
+        };
       });
-    }
+    },
+  },
+  methods: {
+    checkAndSelectSeason(seasonId) {
+      if (this.seasons.some(({ id }) => id === seasonId)) {
+        const [selectedSeason] = this.seasons.filter(({ id }) => id === seasonId);
+        this.selectedSeason = selectedSeason;
+      } else {
+        const [selectedSeason] = this.seasons;
+        this.selectedSeason = selectedSeason;
+      }
+    },
   },
   created() {
     this.loadingData = true;
     io.socket.get('/stats', (res) => {
-      this.seasons = res;
-      const [selectedSeason] = this.seasons
-      this.selectedSeason = selectedSeason;
-      this.loadingData = false;
+      if (res.length > 0) {
+        this.seasons = res;
+        const seasonId = parseInt(this.$route.params.seasonId);
+        this.checkAndSelectSeason(seasonId);
+      }
     });
+    this.loadingData = false;
+  },
+  beforeRouteUpdate(to, from, next) {
+    this.loadingData = true;
+    const seasonId = parseInt(to.params.seasonId);
+    this.checkAndSelectSeason(seasonId);
+    this.loadingData = false;
+    next();
```

Comment:
We very likely need some better way to track loading states since we're doing this all over the place @itsalaidbacklife. I'm thinking at this point some sort of mixin for providing `$isLoading` as a state value to any component that includes it. Doesn't need to be for this PR though.

Reply:
Yeah I think we could also explore the use of the `<Suspense>` tag as a piece of the puzzle ([docs](https://vuejs.org/guide/built-ins/suspense.html)). I think this would largely manifest as a composable, but the trick will be in establishing a uniform api given that different components have different requirements for when loading states should display